sys/net/gnrc/link_layer/netdev2/gnrc_netdev2_eth.c

```diff
@@ -87,7 +87,7 @@ static gnrc_pktsnip_t *_recv(gnrc_netdev2_t *gnrc_netdev2)
         DEBUG("gnrc_netdev2_eth: received packet from %02x:%02x:%02x:%02x:%02x:%02x "
                 "of length %zu\n",
                 hdr->src[0], hdr->src[1], hdr->src[2], hdr->src[3], hdr->src[4],
-                hdr->src[5], nread);
+                hdr->src[5], (size_t) nread);
```

Comment:
For this it might makes sense to actually change the `printf` formatting option.